src/ChainRouter.php

```diff
@@ -64,7 +64,8 @@ class ChainRouter implements ChainRouterInterface, WarmableInterface
      */
     public function __construct(LoggerInterface $logger = null)
     {
-        $this->logger = $logger;
+        $this->logger = $logger;      
```

Comment:
styleci is unhappy because there is trailing whitespace added. the other failures are not related, i will fix those on master.

Reply:
fixed in master. if you remove the trailing whitespace and rebase, it should get green here.

Reply:
I can fix the whitespace errors, regarding the check for null I would suggest just removing it as initializing it in the constructor should take care of the problem of needing it uninitialized.

Another thought (havn't checked) is if the change is needed in DynamicRouter as  well.

Reply:
Hmm I would tend to think the correct solution is to always set the context in all() regardless if it's null or not but I have a hard time getting it to play nicely with the test. Of course the loop could check to see if the context is already the same before trying to set it but that doesn't really seem like the ideal solution either.

I tried to avoid using once() and just comparing the contexts afterwards but the mocks don't keep the values.

I unfortunately don't really have much more time to spend on this (having solved my real problem in another way) and of course you could argue the bug is in localeAwareListener and that there is nothing guaranteeing ->getContext actually returns a non-null object but that is kind of problematic as the standard Router de facto does have that behavior.

(My main obstacle is inexperience with phpunit mock-object's and the structuring of the tests)

I may have time to look at this in the future but not this week.

Thanks anyway for writing good code and keeping up.